### ISSUE TITLE
Siege of Boralus wrong count

### ISSUE THREAD

**Issue by xIGBClutchIx** [Meivyn/AdventureGuideLockouts#1]
In the BFA dungeon Siege of Boralus the count is wrong due to one of the bosses being only able to be killed by the opposite faction. Sergeant Bainbridge (Horde) or Chopper Redhook (Alliance). Intended should be 4/4 but instead saids 4/5.

**Comment by Meivyn** [comment]
Thanks for your report. I'll fix this in the next commit.

**Comment by Meivyn** [comment]
I don't have access to Siege of Boralus yet. Can you do a ``/dump GetInstanceInfo()`` while in the instance and give me the exact string at ``[1]`` please?

**Comment by xIGBClutchIx** [comment]
Yep here you go

```
[1]="Siege of Boralus", 
```

**Comment by Meivyn** [comment]
Thanks. Should be fixed in next release. Just do a ``/reload``.

**Comment by xIGBClutchIx** [comment]
That looks to have fixed the icon but now when I highlight over it the final boss is missing in the tooltip.
I assume looking at your fix it just scans the first 4 bosses but technically there is 5 bosses and wow just does not let you see one and kill it because of opposite faction.

**Comment by Meivyn** [comment]
If you have cleared the instance, it should display the 4 bosses you had killed. Can you give me a screenshot of your issue?

**Comment by xIGBClutchIx** [comment]
A before and after of the fix.

Before latest version:
![before](https://user-images.githubusercontent.com/975065/44342610-97c33f00-a459-11e8-9234-564e4f4610c9.jpg)

After latest version:
![after](https://user-images.githubusercontent.com/975065/44342618-998d0280-a459-11e8-93b4-ba47022c89bd.jpg)

What happens is when you put the maxBosses to 4 it can't get to the last one because it just loops to 4 and the last boss is 5.
https://github.com/Meivyn/EncounterJournalSavedInstances/blob/master/EncounterJournalSavedInstances.lua#L303
The only way I thought of fixing it is putting in special code if a person is alliance or horde.

**Comment by Meivyn** [comment]
I can't test it myself so I'll need your help to try something.
Try to replace the loop by this:
```LUA
local b = 1
while GetSavedInstanceEncounterInfo(i, b) do
	local bossName, _, isKilled = GetSavedInstanceEncounterInfo(i, b)
	table.insert(bosses, {
		name = bossName,
		isKilled = isKilled
	})
	b = b + 1
end
```

**Comment by xIGBClutchIx** [comment]
That looks to have worked.
![test](https://user-images.githubusercontent.com/975065/44346447-ee357b00-a463-11e8-983c-acfc05c21654.jpg)

**Comment by Meivyn** [comment]
Fixed in 5b77394c26a431256e72ff57d5952ad662a276bc.
